SAMstrict.tex

```diff
@@ -0,0 +1,398 @@
+\samstrictrule{Duplicate last segment primary records}{Of the primary records with the same QNAME, at most one can have FLAG 0x80 set}
+\samstrictrule{Missing last segment primary record}{Of the records with the same QNAME with 0x1 FLAG, at least one record must have FLAG 0x80 set.}
+\samstrictrule{Multiple primary alignment records}{Each segment must have at most one record with FLAG 0x100 and 0x800 not set.}{\samrule}
+\samstrictrule{Missing primary alignment record}{Each segment must have at least one record with FLAG 0x100 and 0x800 not set.}{\samrule}
+\samstrictrule{Unknown FLAG bit set}{FLAG bits higher than 0x800 must not be set}{\samrule}
+\samstrictrule{Unmapped reads should not have FLAG 0x10 set}{Unmapped reads should be stored in the orientation in which they came off the sequencing machine and have 0x10 unset.}{\v15bestpractice}
+
+\subsection{RNAME}
+\samstrictrule{Malformed RNAME}{RNAME must conform to the regex {\tt \char92*|[!-()+-\char60\char62-\char126][!-\char126]*}}{\samrule}
+\samstrictrule{RNAME not present in reference}{RNAME must be equal to the value of one of the SQ SN values defined in the header.}
+\samstrictrule{RNAME contains character not supported by VCFv4.3}{RNAME must not contain any of the following characters: {\tt \char60\char62\char91\char93\char58\char42}}{\vcf43}
+\samstrictrule{RNAME not supported by VCFv4.3}{RNAME must be not be one of {\tt DEL}, {\tt INS}, {\tt DUP}, {\tt INV}, {\tt CNV}, or {\tt BND}.}{\vcf43}
+\samstrictrule{RNAME does not match mate}{For a unmapped paired-end or mate-pair read whose mate is mapped, the unmapped read should have RNAME identical to its mate}{\v15bestpractice}
+\samstrictrule{RNAME specified for unmapped template}{If all segments in a template are unmapped, their RNAME should be set as *}{\v15bestpractice}
+
+
+\subsection{POS}
+\samstrictrule{Record placed outside of reference sequence}{If FLAG 0x4 is not set, POS must be between 0 and the length of the RNAME reference sequence inclusive. The length of the RNAME reference sequence can be found in the SQ header LN tag value for the SQ header with a SN tag matching the RNAME.}{\v15bestpractice}
+\samstrictrule{Invalid POS}{POS cannot be 0 if FLAG 0x4 is set.}
+\samstrictrule{Invalid POS}{POS cannot be negative.}{\samrule}
+\samstrictrule{Invalid POS}{POS cannot be greater than 2147483647.}{\samrule}
+\samstrictrule{POS specified without RNAME}{If RNAME is *, POS must be 0.}
+\samstrictrule{POS does not match mate.}{For a unmapped paired-end or mate-pair read whose mate is mapped, the unmapped read should have POS identical to its mate}{\v15bestpractice}
+\samstrictrule{POS specified for unmapped template}{If all segments in a template are unmapped, their POS should be set as 0.}{\v15bestpractice}
+
+\subsection{MAPQ}
+\samstrictrule{Invalid MAPQ}{MAPQ must be between 0 and 255 inclusive.}{\samrule}
+\samstrictrule{Missing MAPQ}{Aligned reads should not have 255 MAPQ.}{\v15bestpractice}
```

Comment:
I have seen unaligned reads with MAPQ 255 before, but not sure about aligned.   The spec doesn't actually care - 255 is simply "unavailable" and has nothing to do with mapped or otherwise, although it does make more sense for mapped data than unmapped data.  Consider: I found an alignment and it's <here>, but I haven't bothered to do any of the maths for likelihood of it being correct nor have I looked for another alignment - hence 255.